parser/parser.go

```diff
@@ -180,6 +181,30 @@ func Normalize(input string) (result string, err error) {
 	return
 }
 
+func Split(input string) (result []string, err error) {
+	inputC := C.CString(input)
+	defer C.free(unsafe.Pointer(inputC))
+
+	resultC := C.pg_query_split_with_parser(inputC)
+	defer C.pg_query_free_split_result(resultC)
+
+	if resultC.error != nil {
+		err = newPgQueryError(resultC.error)
+		return
+	}
+
+	stmts := (**C.PgQuerySplitStmt)(unsafe.Pointer(resultC.stmts))
+	for i := 0; i < int(resultC.n_stmts); i++ {
+		stmtptr := (**C.PgQuerySplitStmt)(unsafe.Pointer(uintptr(unsafe.Pointer(stmts)) + uintptr(i)*unsafe.Sizeof(*stmts)))
+		stmt := **stmtptr
+
+		end := stmt.stmt_location + stmt.stmt_len
+		result = append(result, strings.TrimSpace(input[stmt.stmt_location:end]))
```

Comment:
This is a small detail, but I wonder if we should make trim space optional? (i.e. add an argument `trimSpace` to the function)

If I recall Go's copy on write semantics correctly, if you keep it a simple slice of the input string, that avoids extra allocations, which may be desirable when working with large texts.